src/Telegram/Types/ChatMember.php

```diff
@@ -149,4 +52,30 @@ public function mapSubObjects(string $key, array $data): TelegramTypes
 
         return parent::mapSubObjects($key, $data);
     }
+
+    /**
+     * Create specific ChatMember class based on 'status'. If there is no match,
+     */
+    public static function create(array $data, LoggerInterface $logger): ChatMember
+    {
+        $status = $data['status'] ?? null;
+
+        foreach (self::CHILD_CLASS_REFERENCES as $childClassReference) {
+            if ($status === $childClassReference::STATUS) {
+                return new $childClassReference($data, $logger);
+            }
+        }
```

Comment:
While this is pleasant to look at, I much prefer dropping the constant within each type in order to keep those objects as clean as possible, and make the factory call the child classes directly. 

I'm not a big fan of this, but I believe that's the best way, very much as we do here: https://github.com/unreal4u/telegram-api/blob/master/src/Telegram/Types/Message.php#L501-L566 (Yes, I hate that part of the code but it was the only way I could think of at the time).